internal.nix

```diff
@@ -127,7 +177,10 @@ rec {
     if dependency ? resolved && dependency ? integrity then
       dependency // { resolved = "file://" + (toString (fetchurl (makeSourceAttrs name dependency))); }
     else if dependency ? from && dependency ? version then
-      makeGithubSource sourceHashFunc name dependency
+      if lib.hasPrefix "github:" dependency.version then
```

Comment:
We should start adding comments here which cases these cover. This function is getting a bit out of hand. Can you add an example for these two cases as one-line comment?